Kernel/Signature.hpp

```diff
@@ -884,9 +884,9 @@ class Signature
   unsigned formulaCount(Term* t);
 
 
-  bool isTermAlgebraSort(TermList sort) { return _termAlgebras.find(sort); }
-  Shell::TermAlgebra *getTermAlgebraOfSort(TermList sort) { return _termAlgebras.get(sort); }
-  void addTermAlgebra(Shell::TermAlgebra *ta) { _termAlgebras.insert(ta->sort(), ta); }
+  bool isTermAlgebraSort(TermList sort) { return sort.isTerm() && _termAlgebras.find(sort.term()->functor()); }
+  Shell::TermAlgebra *getTermAlgebraOfSort(TermList sort) { return _termAlgebras.get(sort.term()->functor()); }
+  void addTermAlgebra(Shell::TermAlgebra *ta) { _termAlgebras.insert(ta->sort().term()->functor(), ta); }
```

Comment:
What is the reason for these changes?

Reply:
Since datatypes can be polymorphic now, it's easier to track them via their sort functor (otherwise we have to match sorts when we want to get the term algebra corresponding to them).

Reply:
Right, makes sense.